python/pyspark/sql/_typing.pyi

```diff
@@ -37,6 +37,10 @@ from pyspark.profiler import CodeMapDict
 import pyspark.sql.types
 from pyspark.sql.column import Column
 
+__all__ = ["Concatenate"]
```

Comment:
why do we need this `__all__`?

Reply:
This is to suppress the mypy error:

```sh
pyspark/sql/dataframe.py:53: error: Module "pyspark.sql._typing" does not explicitly export attribute "Concatenate"  [attr-defined]
```

`Concatenate` is now being imported into `python/pyspark/sql/dataframe.py` from its corresponding `_typing.pyi` file, so it appears it now needs to be explicitly exported from `_typing.pyi`.